[CT-2846] Update dependency pins: `click<9` + `sqlparse<0.5`

Unpin `click` now that a new version has been released that _should_ fix the typing issues:
- https://github.com/dbt-labs/dbt-core/issues/8048
- https://github.com/pallets/click/issues/2558#issuecomment-1640926299

---

We removed the `sqlparse` pin in https://github.com/dbt-labs/dbt-core/pull/7993 — but we should still have an upper bound since it's major-0 software. Previously we had the upper bound pinned to `<0.5`.

---

We should backport to `1.6.latest` (for inclusion in v1.6.0rc2/final) + `1.5.latest` (for inclusion in v1.5.4)

opening up the pin for click does still have some mypy failures.  specifically these just 3 in #8232.  Question is if we want to ignore them or just not upgrade yet.

```
core/dbt/cli/params.py:392: error: Argument 2 to "option" has incompatible type "**Dict[str, Union[Type[Tuple[Any, ...]], bool, Type[MultiOption], str, None]]"; expected "Optional[Type[Option]]"  [arg-type]
core/dbt/cli/params.py:393: error: Argument 2 to "option" has incompatible type "**Dict[str, Union[Type[Tuple[Any, ...]], bool, Type[MultiOption], str, None]]"; expected "Optional[Type[Option]]"  [arg-type]
core/dbt/cli/params.py:394: error: Argument 3 to "option" has incompatible type "**Dict[str, Union[Type[Tuple[Any, ...]], bool, Type[MultiOption], str, None]]"; expected "Optional[Type[Option]]"  [arg-type]
```

--- a/.changes/unreleased/Dependencies-20230727-145726.yaml
+++ b/.changes/unreleased/Dependencies-20230727-145726.yaml
@@ -1,0 +1,6 @@
+kind: Dependencies
+body: Add upper bound to sqlparse pin of <0.5
+time: 2023-07-27T14:57:26.40416-05:00
+custom:
+  Author: emmyoop
+  PR: "8236"

--- a/core/setup.py
+++ b/core/setup.py
@@ -69,8 +69,7 @@ setup(
         "pathspec>=0.9,<0.12",
         "isodate>=0.6,<0.7",
         # ----
-        # There was a pin to below 0.4.4 for a while due to a bug in Ubuntu/sqlparse 0.4.4
-        "sqlparse>=0.2.3",
+        "sqlparse>=0.2.3,<0.5",
         # ----
         # These are major-version-0 packages also maintained by dbt-labs. Accept patches.
         "dbt-extractor~=0.5.0",

